### PR TITLE
Feature/es6 features

### PR DESCRIPTION
**Pull request by NJaryal**
Generate Function

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -18,40 +18,40 @@ const authorNames = () => {
...
+    h4.innerHTML = `AUTHOR: ${news.author}`;
+    strong.innerHTML = `Title: ${news.title} `;
+    p.innerHTML = `Description: ${news.description}`;
+    append(li, img);
+    append(li, div);
+    append(div, h4);
+    append(div, strong);
+    append(div, p);        
+    append(ul , li); 
+    spinner.style.display = 'none';
```
Comment (byslavik) [inline script.js]: Lets create a separate class for Spinner

File: script.js [hunk 2]
```diff
@@ -18,40 +18,40 @@ const authorNames = () => {
     let authors = new Proxy({}, validator);
 }
 
-const api = (source) => fetch(`${BASE_URL}`+source+`&apiKey=${API_KEY}`); //Fetch api function
+const api = (source) => fetch(`${BASE_URL}${source}&apiKey=${API_KEY}`); //Fetch api function
+//Render News Function for creation of DOM
+const renderNewsItem = (news) => {
```
Comment (byslavik) [inline script.js]: And I think will be better to have it as a method of NewsChannel class

**Comment by byslavik** [comment]
Also please pay attention to improvements that I sent to you.

File: script.js (hunk 2 above)
Comment (NJaryal) [reply script.js]: Agreed and Applied.

File: src/app.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
+/* import { API_KEY as key } from './variables.js'; //Import and Rename
+console.log(key); */
+let spinner = document.querySelector(".spinner"); //Spinner to show when fetching api data
```
Comment (byslavik) [inline src/app.js]: If we are using Spinner as a class and other helpers as methods, I wouuld say we need to remove it from here

File: src/app.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+  init() {  
+    this.channels = this.sources.reduce((acc, { source, inst }) => {
+      acc[inst] = new NewsChannel(source, this);
+      return acc
+    }, {})
+    
+    this.main.addEventListener('click', this.handleMainClick.bind(this))    
+    this.sources.forEach((item)=>{
+      this.main.innerHTML +=
```
Comment (byslavik) [inline src/app.js]: That could be moved to the separate method

File: src/app.js (hunk 3 above)
Comment (NJaryal) [reply src/app.js]: Removed redundant code

File: src/app.js (hunk 4 above)
Comment (NJaryal) [reply src/app.js]: Created separate initialContent method

File: client/src/app.js [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+        return await response.json();
+      } catch (error) {
+        console.log(error);
+      }    
+    }   
+    
+  handleMainClick({ target }) {
+      if(target.tagName.toLowerCase() === 'a' && target.dataset.loadChannel) {
+        const channelName = target.dataset.loadChannel;
+        console.log(channelName);
```
Comment (byslavik) [inline client/src/app.js]: lets remove that console.log

File: client/src/app.js [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+const settings = {
+  spinnerSelector: '.spinner',
+  ulSelector: '#newsArticlesList',
+  gridSection: '.newsSection',
+  channelUIListsSelector: '.thumbnail',
+  BASE_URL,
+  API_KEY
+}
+class Spinner {
```
Comment (byslavik) [inline client/src/app.js]: could you please split you file into small pieces and use import/export to make them a little bit more readable?

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        path: path.resolve(__dirname, 'dist')         
+    },    
+    module: {
+        rules: [
+            {   test: '/\.m?js$/',
+                exclude: /(node_modules)/,
+                use: {
+                    loader: 'babel-loader',
+                    options: {
+                        presets: ['@babel/preset-env']
```
Comment (byslavik) [inline webpack.config.js]: Lets specify targets for that preset
https://babeljs.io/docs/en/babel-preset-env

File: .babelrc [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["@babel/preset-env"]
```
Comment (byslavik) [inline .babelrc]: if you are using .babelrc you should not include it in webpack.config

File: client/src/app.js (hunk 5 above)
Comment (NJaryal) [reply client/src/app.js]: I kept it purposefully - As we have to remove the consoles from webpack.

File: client/src/app.js (hunk 6 above)
Comment (NJaryal) [reply client/src/app.js]: I separated the classes into modules

File: webpack.config.js (hunk 7 above)
Comment (NJaryal) [reply webpack.config.js]: Either Browser perspective or es6 perspective ?

"targets": {
          "esmodules": true
        }

**Pull request merged**